### PR TITLE
Feature/new window right click

### LINKED ISSUE #2000: [Feature Request]: right-click menu to open a channel or video in a new window
### Guidelines

- [X] I have searched the [issue tracker](https://github.com/FreeTubeApp/FreeTube/issues) for a feature request that matches the one I want to file, without success.

### Problem Description

Sometime I want to open a video/channel to a new window to check it later. Right now I have to click on the new window button and copy the video/channel name into the search bar to.

### Proposed Solution

A right-click menu to open the video/channel to a new window would be an ideal solution.

### Alternatives Considered

a middle click would also be great

### Issue Labels

ease of use improvement

### Additional Information

It's related to #333 but I don't think it's the same

### PR DESCRIPTION
**Pull request by constraintAutomaton**
---
Feature/new window right click
---

**Pull Request Type**

- [ ] Bugfix
- [x] Feature Implementation

**Related issue**
closes #2000 

**Description**
When right-clicking a link, a video or a channel image a context menu appears giving the option to open the source into a new window.

**Screenshots**
Screenshot from 2022-01-25 18-02-13

Screenshot from 2022-01-25 18-02-47

Screenshot from 2022-01-25 18-02-56

Screenshot from 2022-01-25 18-07-48

Screenshot from 2022-01-25 18-07-55

**Testing**
Right click on a link or video and check if it is open in a new window.

**Desktop**
- OS: Ubuntu
- OS Version: 20.04.3 LTS
- FreeTube version: v0.15.1 Beta

**Additional context**
- enters in interactions with #1895
- I've changed the way the navigation between some pages is implemented

### CODE REVIEW

**Comment by PikachuEXE** [comment]
I think this PR will conflict with `createWindow` change in #1895 
as I updated that function to accept arguments in object

Not sure why that PR takes so long to be reviewed :S

**Comment by constraintAutomaton** [comment]
> I think this PR will conflict with `createWindow` change in #1895 as I updated that function to accept arguments in object
> 
> Not sure why that PR takes so long to be reviewed :S

when it will be merged I will use your more comprehensive/elegant implementation of `createWindow`

**Comment by PikachuEXE** [comment]
I prefer that style much more than positional arguments especially when there are many arguments / optional arguments
https://simonsmith.io/destructuring-objects-as-function-parameters-in-es6

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
>When i middle click on an yt url it doesnt open a new window. Would like to see that instead of opening it in the same window.

Maybe it's too much to ask in this PR but is it possible for u to fix this behavior in this PR. Closes https://github.com/FreeTubeApp/FreeTube/issues/1923 (fixing this PR is partially covered by Pikachu in #1895 )

**Comment by constraintAutomaton** [comment]
> > When i middle click on an yt url it doesnt open a new window. Would like to see that instead of opening it in the same window.
> 
> Maybe it's too much to ask in this PR but is it possible for u to fix this behavior in this PR. Closes #1923 (fixing this PR is partially covered by Pikachu in #1895 )

I can fix it too, I'm not sure I understand what is in parathesis.

**Comment by PikachuEXE** [comment]
I am developing the fix already, figuring out how to handle `query` on new window
That fix deserves a separate PR
Also that fix I am developing depends on #1895

**Comment by PikachuEXE** [comment]
A bigger problem is a few PR stuck by not having a 3rd reviewer =_=|||

**Comment by constraintAutomaton** [comment]
> I am developing the fix already, figuring out how to handle `query` on new window That fix deserves a separate PR Also that fix I am developing depends on #1895

ok then I let you do it.

**Comment by PikachuEXE** [comment]
> When i middle click on an yt url it doesnt open a new window.

@efb4f5ff-1298-471a-8973-3d47447115dc @constraintAutomaton 
Fix submitted as #2083 
Test it with more types of links though

**Review by efb4f5ff-1298-471a-8973-3d47447115dc — DISMISSED** [review]
Everything seems to work fine. LGTM

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
closes #884

**Comment by PikachuEXE** [comment]
@efb4f5ff-1298-471a-8973-3d47447115dc 
I guess you mean #2083?

Kind of confusing when I first read your last message here...

**Comment by PikachuEXE** [comment]
Will review remaining PR this week... or next
Busy!!

File: src/main/index.js
Comment (PikachuEXE) [inline src/main/index.js:33]: Is this accurate enough?

**Review by PikachuEXE — COMMENTED** [review]
Tested functions locally and seems fine
I only have 1 doubt, see my line comment

File: src/main/index.js
Comment (constraintAutomaton) [reply src/main/index.js:33]: I think so, because any localhost link would be an in app page that the user might want to open in a new window.

File: src/renderer/components/side-nav/side-nav.js
Comment (PrestonN) [inline src/renderer/components/side-nav/side-nav.js:81]: Please double check and see if this can actually be removed. I believe the `@keypress` action is here so that the sideNav works via accessibility controls (Ex: Using Tab to navigate through the menu and hitting enter to navigate to that channel). If it's still needed for this workflow, then please re-add this to the file.

File: src/main/index.js
Comment (PrestonN) [reply src/main/index.js:33]: My only hesitation with this method is that we'd one day like to have a version of FreeTube for the web and going this route would basically make this feature non-existent in a web based version. If there's an alternative then I'd much rather go that route, but if there isn't anything then this is fine for now.

**Review by PrestonN — CHANGES_REQUESTED** [review]
Please see my comments. Otherwise looks good.

File: src/main/index.js
Comment (ChunkyProgrammer) [reply src/main/index.js:33]: a library like [this](https://github.com/johndatserakis/vue-simple-context-menu)(maybe see if there's a better alternative) would allow using a different context menu for both web and electron as it is a vue component. Bonus: using a vue library like this could fix the issue with #2479 in the future

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
@constraintAutomaton could u provide an update please?

**Comment by constraintAutomaton** [comment]
> @constraintAutomaton could u provide an update please?

I'm gonna finish it this week, sorry, I got very busy.

File: src/main/index.js
Comment (constraintAutomaton) [reply src/main/index.js]: now I detect the address of the website

**Comment by constraintAutomaton** [comment]
@efb4f5ff-1298-471a-8973-3d47447115dc 

it's done, but I don't understand how to fix Lint code Base github action

**Comment by ChunkyProgrammer** [comment]
> @efb4f5ff-1298-471a-8973-3d47447115dc
> 
> it's done, but I don't understand how to fix Lint code Base github action

we'll be updating that action soon. Don't worry about making it pass for the time being

**Review by PikachuEXE — APPROVED** [review]
Tested locally

**Comment by PikachuEXE** [comment]
@ChunkyProgrammer @efb4f5ff-1298-471a-8973-3d47447115dc @absidue ?

**Review by efb4f5ff-1298-471a-8973-3d47447115dc — DISMISSED** [review]

File: src/main/index.js
Comment (efb4f5ff-1298-471a-8973-3d47447115dc) [reply src/main/index.js]: @MarmadileManteater any thoughts on this

**Review by absidue — DISMISSED** [review]

**Comment by MarmadileManteater** [comment]
By making all of the channel names in the slide out menu into anchors, the text color and text-style changed. This could be fixed with a little CSS.
0.17.1v:
image
This PR:

**Comment by PikachuEXE** [comment]
Just fixed it

**Comment by constraintAutomaton** [comment]
> Just fixed it

thank you!

**Review by efb4f5ff-1298-471a-8973-3d47447115dc — APPROVED** [review]

**Comment by PikachuEXE** [comment]
@ChunkyProgrammer @absidue Please review 👋

**Comment by absidue** [comment]
@PikachuEXE we need @PrestonN to approve (even if we approve it won't be merged until he approves it as he requested changes)

**Comment by PikachuEXE** [comment]
Try approving and see if it's auto merged
I don't see change requested on reviewer list
image

**Review by ChunkyProgrammer — APPROVED** [review]
It already had 3 reviews at one point so like absidue said, it's probably not gonna merge

**Comment by constraintAutomaton** [comment]
Will he comeback "soon" or should I close the PR and create a new one?

**Pull request merged**
